rust/src/transformations/make_stable_expr/expr_clip/mod.rs

```diff
@@ -0,0 +1,158 @@
+use polars::prelude::*;
+use polars_plan::dsl::{Expr, FunctionExpr};
+
+use crate::core::{ExprFunction, Function, MetricSpace, StabilityMap, Transformation};
+use crate::domains::{AtomDomain, Bounds, ExprDomain, NumericDataType, SeriesDomain};
+use crate::error::*;
+use crate::traits::Number;
+
+use super::{DatasetOuterMetric, StableExpr};
+
+/// Make a Transformation that returns a `clip(bounds)` expression for a LazyFrame.
+///
+/// # Arguments
+/// * `input_domain` - Expr domain
+/// * `input_metric` - The metric under which neighboring LazyFrames are compared
+/// * `expr` - The clipping expression
+pub fn make_expr_clip<MS, MI>(
+    input_domain: ExprDomain,
+    input_metric: MS,
+    expr: Expr,
+) -> Fallible<Transformation<ExprDomain, ExprDomain, MS, MI>>
+where
+    MS: 'static + DatasetOuterMetric,
+    MI: 'static + DatasetOuterMetric,
+    MI::Distance: Clone,
+    (ExprDomain, MS): MetricSpace,
+    (ExprDomain, MI): MetricSpace,
+    Expr: StableExpr<MS, MI>,
+{
+    let Expr::Function {
+        input, function, ..
+    } = expr
+    else {
+        return fallible!(MakeTransformation, "expected function expression");
+    };
+
+    let FunctionExpr::Clip { has_min, has_max } = function else {
+        return fallible!(MakeTransformation, "expected clip function");
+    };
+
+    if !has_min || !has_max {
+        return fallible!(MakeTransformation, "Clip must have min and max");
+    }
+
+    let [input, lower, upper] = <[Expr; 3]>::try_from(input)
+        .map_err(|_| err!(MakeTransformation, "Clip expects 3 arguments"))?;
+
+    let t_prior = input.make_stable(input_domain.clone(), input_metric.clone())?;
+    let (input_domain, input_metric) = t_prior.output_space();
+
+    let mut output_domain = t_prior.output_domain.clone();
+    let (lower, upper) = {
+        let series_domain = output_domain.active_series_mut()?;
+
+        match &series_domain.field.dtype {
+            DataType::UInt32 => extract_bounds::<u32>(lower, upper, series_domain)?,
+            DataType::UInt64 => extract_bounds::<u64>(lower, upper, series_domain)?,
+            DataType::Int8 => extract_bounds::<i8>(lower, upper, series_domain)?,
+            DataType::Int16 => extract_bounds::<i16>(lower, upper, series_domain)?,
+            DataType::Int32 => extract_bounds::<i32>(lower, upper, series_domain)?,
+            DataType::Int64 => extract_bounds::<i64>(lower, upper, series_domain)?,
+            DataType::Float32 => extract_bounds::<f32>(lower, upper, series_domain)?,
+            DataType::Float64 => extract_bounds::<f64>(lower, upper, series_domain)?,
+            _ => return fallible!(MakeTransformation, "unsupported dtype"),
+        }
+    };
+
+    let series_name = input_domain.active_series()?.field.name.as_str();
+
+    // Margins on the active_column could be preserved but this functionality has not been implemented yet
```

Comment:
I don't see `active_column` in this PR, and it doesn't seem to be a polars thing either. Link to an issue with more context? Seems like it's something you'd like to do, and we don't want to lose track.